pkg/utils/validation/admissionplugins/admissionplugins.go

```diff
@@ -179,9 +180,11 @@ func ValidateAdmissionPlugins(admissionPlugins []core.AdmissionPlugin, version s
 		if err != nil {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("name"), plugin.Name, err.Error()))
 		} else if !supported {
-			// If the plugin is not supported, but it's disabled and is a plugin in migration, then skip it.
-			if pointer.BoolDeref(plugin.Disabled, false) && pluginsInMigration.Has(plugin.Name) {
-				continue
+			// If the plugin is not supported, but it's disabled and it's a plugin in migration, then skip it.
+			if constraint, ok := PluginsInMigration[plugin.Name]; ok {
+				if constraint.Check(semver.MustParse(version)) && pointer.BoolDeref(plugin.Disabled, false) {
```

Comment:
Nit to reduce nesting

```suggestion
			if constraint, ok := PluginsInMigration[plugin.Name]; ok && constraint.Check(semver.MustParse(version)) && pointer.BoolDeref(plugin.Disabled, false) {
```